app/src/component/mediaPlayer/view.js

```diff
@@ -284,9 +285,18 @@ class MediaPlayer extends React.PureComponent {
   }
 
   getEncodedDownloadPath = (fileInfo) => {
+    if (this.state.encodedFilePath) {
+      return this.state.encodedFilePath;
+    }
+
     const { file_name: fileName } = fileInfo;
     const encodedFileName = encodeURIComponent(fileName).replace(/!/g, '%21');
-    return fileInfo.download_path.replace(new RegExp(fileName, 'g'), encodedFileName);
+    const re = new RegExp(fileName.replace(/\-/g, '\\-')
+                                  .replace(/\(/g, '\\(')
+                                  .replace(/\)/g, '\\)'), 'g');
+    const encodedFilePath = fileInfo.download_path.replace(re, encodedFileName);
```

Comment:
Not exactly sure what this is doing. `re` would just be a string since you are using `.replace`. Should `re` be a regex pattern?

Reply:
This makes sense and makes this seem like overengineering the solution in retrospect. I can't even remember why I used `RegExp` in the first place.